tests/test_testing.py

```diff
@@ -319,3 +321,22 @@ def cli(f):
 
     result = runner.invoke(cli, ["-"])
     assert result.output == "<stdin>\nr"
+
+
+def test_isolated_runner(runner):
+    lingering_directory = None
+    with runner.isolated_filesystem() as directory:
+        lingering_directory = directory
+        assert os.path.exists(directory)
+
+    assert not os.path.exists(lingering_directory)
+
+
+def test_isolated_runner_custom_tempdir(runner):
+    lingering_directory = None
+    with runner.isolated_filesystem(tempdir=tempfile.gettempdir()) as directory:
+        lingering_directory = directory
+        assert os.path.exists(directory)
+
+    assert os.path.exists(lingering_directory)
+    shutil.rmtree(lingering_directory)
```

Comment:
Looking good!

One thing to notice: `shutil.rmtree` won't be called if any of the code above raises an exception, meaning that this code might or might not leave an artifact lying around. A context manager could fix this, as could a `try`/`finally` block.

However, in this case, it's not a big deal since `tempfile.gettempdir` will use the OS's temporary directory by default.